src/sentry/incidents/models.py

```diff
@@ -500,6 +532,18 @@ def save(self, **kwargs: Any) -> None:
         self._validate_actor()
         return super().save(**kwargs)
 
+    @register_alert_subscription_update(AlertRuleMonitorType.ACTIVATED)
+    def clean_expired_alerts(self, subscription: QuerySubscription) -> bool:
```

Comment:
I'm not totally sure that this works... This expects `self` to be passed, but you're only passing subscription. Does this actually need to live on `IncidentProject`?

Reply:
Hmm, in testing this works as expected with the invocation. I'm also working on adding a test to make sure `invoke_alert_subscription_update` is called as expected which can also confirm that.

I'm not sure the relationship of an `IncidentProject`, so maybe? My thought was that the `AlertRule` is the thing that wants to update when there's a subscription change, so having that live on the model associated with where the type is used made the most sense to me. fwiw, i could also pass the alert rule instance into the function (i tried doing that initially thinking the same thing, but it seemed like i didn't have to)

(in general it's a little unclear to me how incidents map to AlertRules. i would expect an alert rule to create an incident when a threshold is met)

Reply:
ooohhhh @wedamija was this an unfortunate fold in github?